src/python/pants/backend/codegen/protobuf/python/python_protobuf_subsystem.py

```diff
@@ -59,9 +61,42 @@ def runtime_dependencies(self) -> UnparsedAddressInputs:
     def mypy_plugin(self) -> bool:
         return cast(bool, self.options.mypy_plugin)
 
-    @property
-    def mypy_plugin_version(self) -> str:
-        return cast(str, self.options.mypy_plugin_version)
+
+class PythonProtobufMypyPlugin(PythonToolRequirementsBase):
+    options_scope = PythonProtobufSubsystem.subscope("mypy-plugin")
+    help = (
+        "Configuration of the mypy-protobuf type stub generation plugin for the Protobuf Python "
+        "backend."
+    )
+
+    default_version = "mypy-protobuf==2.4"
+    register_interpreter_constraints = True
+    default_interpreter_constraints = ["CPython>=3.6"]
+
+    def plugin_requirement(self, python_protobuf_subsystem: PythonProtobufSubsystem) -> str:
+        deprecated_key = "mypy_plugin_version"
+        mypy_plugin_requirement = python_protobuf_subsystem.options.get(deprecated_key)
+        if mypy_plugin_requirement:
+            # TODO(John Sirois): When removing this deprecation, its safe to remove the
+            #  `--mypy-plugin-version` option registration in PythonProtobufSubsystem above since
+            #  the `version` option registered by this subsystem is in a subscope picked to shadow
+            #  the deprecated option in flag and environment variable namespaces; i.e.: both
+            #  --python-protobuf-mypy-plugin-version and PANTS_PYTHON_PROTOBUF_MYPY_PLUGIN_VERSION
+            #  will continue to work."
+            deprecated_conditional(
```

Comment:
I'm not sure that I fully understand how this deprecation works, but given that the plugin was added recently, I won't investigate too closely.

Reply:
Yeah, it's tricky enough that I went through the steps to test all this. The only deprecated thing is the config syntax and this triggers only for that by design.